src/lakefs_spec/util.py

```diff
@@ -58,8 +89,9 @@ def parse(path: str) -> tuple[str, str, str]:
     # https://docs.lakefs.io/understand/model.html#repository
     # Second regex is the branch: Only letters, digits, underscores
     # and dash, no leading dash
+    path = PathHandler(path)
```

Comment:
Question: How do we intend to use the `parse`  (and `md5_checksum`) method in other places?
If we don't we could directly enforce a `PathHandler` type and remove the typecasting in the first line of the function.

Reply:
I think at most, users will pass path-like objects for the `lpath` argument, as the `rpath` argument is not really that interesting to wrap into a path (the methods on it don't work for a remote path, which is what `universal_pathlib` is trying to fix).

So we can leave those interfaces as is, and just coerce directly before. (That being said, `lpath` should be a `str | Path` on `md5_checksum`.)